test/cpp/end2end/xds/xds_override_host_end2end_test.cc

```diff
@@ -136,42 +160,112 @@ TEST_P(OverrideHostTest, HappyPath) {
   WaitForAllBackends(DEBUG_LOCATION);
   // First call gets the cookie. RR policy picks the backend we will use.
-  auto server_initial_metadata =
-      SendRpcGetServerMetadata(DEBUG_LOCATION, RpcOptions());
-  std::vector<std::string> stateful_session_cookie =
-      GetStatefulSessionCookie(server_initial_metadata);
-  ASSERT_EQ(stateful_session_cookie.size(), 1);
-  size_t backend_idx = -1;
-  for (size_t i = 0; i < backends_.size(); ++i) {
-    if (backends_[i]->backend_service()->request_count() == 1) {
-      backend_idx = i;
-      break;
-    }
-  }
-  ASSERT_NE(-1, backend_idx);
-  ResetBackendCounters();
-  std::vector<std::pair<std::string, std::string>> session_cookie = {
-      {"cookie",
-       absl::StrFormat("%s=%s", kCookieName, stateful_session_cookie[0])}};
+  auto session_cookie = GetAffinityCookieHeaderForBackend(DEBUG_LOCATION, 0);
+  ASSERT_TRUE(session_cookie.has_value());
   // All requests go to the backend we specified
-  CheckRpcSendOk(DEBUG_LOCATION, 5, RpcOptions().set_metadata(session_cookie));
-  EXPECT_EQ(backends_[backend_idx]->backend_service()->request_count(), 5);
+  CheckRpcSendOk(DEBUG_LOCATION, 5, RpcOptions().set_metadata(*session_cookie));
+  EXPECT_EQ(backends_[0]->backend_service()->request_count(), 5);
   // Round-robin spreads the load
   ResetBackendCounters();
-  CheckRpcSendOk(DEBUG_LOCATION, backends_.size() * 2);
+  CheckRpcSendOk(DEBUG_LOCATION, 4);
   EXPECT_EQ(2, backends_[0]->backend_service()->request_count());
   EXPECT_EQ(2, backends_[1]->backend_service()->request_count());
   // Call a different service with the same cookie
   ResetBackendCounters();
   CheckRpcSendOk(DEBUG_LOCATION, 5,
                  RpcOptions()
-                     .set_metadata(session_cookie)
+                     .set_metadata(*session_cookie)
                      .set_rpc_service(RpcService::SERVICE_ECHO2));
-  EXPECT_EQ(backends_[backend_idx]->backend_service2()->request_count(), 5);
+  EXPECT_EQ(backends_[0]->backend_service2()->request_count(), 5);
+}
+
+TEST_P(OverrideHostTest, DrainingIncludedFromOverrideSet) {
+  CreateAndStartBackends(3);
+  Cluster cluster = default_cluster_;
+  auto* lb_config = cluster.mutable_common_lb_config();
+  auto* override_health_status_set = lb_config->mutable_override_host_status();
+  override_health_status_set->add_statuses(HealthStatus::HEALTHY);
+  override_health_status_set->add_statuses(HealthStatus::UNKNOWN);
+  override_health_status_set->add_statuses(HealthStatus::DRAINING);
+  balancer_->ads_service()->SetCdsResource(cluster);
+  SetListenerAndRouteConfiguration(balancer_.get(),
+                                   BuildListenerWithStatefulSessionFilter(),
+                                   default_route_config_);
+  balancer_->ads_service()->SetEdsResource(BuildEdsResource(
+      EdsResourceArgs({{"locality0",
+                        {CreateEndpoint(0, HealthStatus::HEALTHY),
+                         CreateEndpoint(1, HealthStatus::UNKNOWN)}}})));
```

Comment:
Let's make both backends HEALTHY here.  We've already tested UNKNOWN in the previous test, so we don't need that extra variable here, since it's not relevant to the test.